gspread/worksheet.py

```diff
@@ -1349,12 +1355,31 @@ def insert_row(self, values, index=1, value_input_option=ValueInputOption.raw):
             should be interpreted. Possible values are ``ValueInputOption.raw``
             or ``ValueInputOption.user_entered``.
             See `ValueInputOption`_ in the Sheets API.
+        :param bool inherit_from_before: (optional) If True, the new row will
+            inherit its properties from the previous row. Defaults to False,
+            meaning that the new row acquires the properties of the row
+            immediately after it.
+            .. warning::
+               `inherit_from_before` must be False when adding a row to the top
+               of a spreadsheet (`index=1`), and must be True when adding to
+               the bottom of the spreadsheet.
```

Comment:
I believe you need to have a blank line before "warning" and immediately after too.